concent_api/api-integration-multi-threads-test.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import hashlib
+import os
+import sys
+import time
+from datetime import datetime
+from threading import Thread
+
+import requests
+from golem_messages.shortcuts import dump
+
+from api_testing_common import api_request
+from api_testing_common import assert_condition
+from api_testing_common import count_fails
+from api_testing_common import get_force_get_task_result
+from api_testing_common import REQUESTOR_PRIVATE_KEY
+from api_testing_common import run_tests
+from common.helpers import get_current_utc_timestamp
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "concent_api.settings")
+
+now = datetime.now()
+testing_subtask_id = str(now.year) + str(now.month) + str(now.day) + str(now.hour) + str(now.minute) + str(
+    now.second) + str(now.microsecond)
+
+responses = []  # type: str
+
+
+def multi_threads(func):
+    def wrapper(*args, **kwargs):
+        for i in range(3):
+            t = Thread(target=func, args=(*args,), kwargs={**kwargs})
+            t.start()
+
+    return wrapper
+
+
+@count_fails
+def test_case_that_multiple_requests_by_one_subtask_will_not_be_cause_of_server_error(cluster_consts, cluster_url, test_id):
+
+    send_correct_request(cluster_consts=cluster_consts, cluster_url=cluster_url)
+
+    time.sleep(4)
+    print('Responses = ' + str(responses))
```

Comment:
I think you forget to assign to responses in the call above ?

Reply:
I'm not sure I understand the question.. I need only names of messages, whole messages are not necessary. In this line I print responses only for someone, who will use this test to see and remember, which messages are expected

Reply:
Yes, I mean the whole flow of this. You create a list outside of this function, then add items inside the function, and then check list outside of the function again. What I am asking is if the function can't return the value that will be added to the list directly ?

Reply:
It is because of multi-threading. I couldn't get response directly from other threads, but it is possible to create list with access from all threads. Each thread saves it's response to this list and it is checked in main thread.